mvc/overview.rst

```diff
@@ -1,8 +1,224 @@
+
+- :doc:`Controllers </controllers/index>` process incoming requests, work with the model, and ultimately return a view to the client that displays the user interface.
+
+This delineation of responsibilities helps you scale the application in terms of complexity because it’s much easier to code, debug, and test something (model, view, or controller) that has a single job rather than something that performs multiple tasks.
+
+Features
+--------
+
+ASP.NET MVC provides the following features:
+  - `Routing`_
+  - `Model binding`_
+  - `Request validation`_
+  - `Razor view engine`_
+  - `Strongly typed views`_
+  - `HTML Helpers`_
+  - `Tag Helpers`_
+  - `View Components`_
+  - `Dependency Injection`_
+  - `Filters`_
+  - `Areas`_
+  - `Mobile development`_
+  - `Web UI and Web APIs`_
+  - `Test-driven Development`_
+  - `Built on top of ASP.NET 5`_
+  - `Tooling`_
+
+Routing
+-------
+
+MVC is built on :doc:`ASP.NET Routing </controllers/routing>`, a powerful URL-mapping component that lets you build applications that have comprehensible and searchable URLs. This enables you to define your application's URL naming patterns that work well for search engine optimization (SEO) and for link generation. You can define your routes using a convenient route template syntax that supports route value constraints, defaults and optional values.
+
+*Convention-based routing* enables you to globally define the URL formats that your application accepts and how each of those formats maps to a specific action method on given controller. When an incoming request is received, the routing engine parses the incoming URL and matches it to one of the defined URL formats, and calls the associated controller's action method. 
+
+.. code-block:: c#
+
+  routes.MapRoute(name: "Default", template: "{controller=Home}/{action=Index}/{id?}");
+
+*Attribute routing* enables you to specify routing information by decorating your controllers and actions with attributes that define your application's routes. This means that your route definitions are placed next to the controller and action with which they're associated.
+
+.. code-block:: c#
+  :emphasize-lines: 1,4
+
+  [Route("api/[controller]")]
+  public class ProductsController : Controller
+  {
+    [HttpGet("{id}")]
+    public IActionResult GetProduct(int id)
+    {
+      ...
+    }
+  }
+
+Model binding
+-------------
+
+MVC :doc:`model binding </models/model-binding>` converts request data (form values, route data, query string parameters, HTTP headers) into objects that the controller can handle. As a result, your controller logic doesn't have to do the work of figuring out the incoming request data; it simply has the data as parameters to its action methods.
+
+.. code-block:: C#
+
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null) { ... }
```

Comment:
This sample might be more easily understood if instead of taking in a complex `LoginViewModel` type it took in `string username, string password, bool rememberMe, string returnUrl)`.